tests/unit/forklift/test_legacy.py

```diff
@@ -3282,18 +3313,55 @@ def test_upload_succeeds_creates_release(
                 release.project.name,
                 release.version,
                 "new release",
-                user,
+                identity if test_with_user else None,
                 db_request.remote_addr,
             ),
             (
                 release.project.name,
                 release.version,
                 f"add source file {filename}",
-                user,
+                identity if test_with_user else None,
                 db_request.remote_addr,
             ),
         ]
 
+        # Ensure that all of our events have been created
+
+        assert record_event.calls == [
+            pretend.call(
+                mock.ANY,
+                tag=EventTag.Project.ReleaseAdd,
+                ip_address=mock.ANY,
+                additional={
+                    "submitted_by": identity.username
+                    if test_with_user
+                    else "OpenID created token",
+                    "canonical_version": release.canonical_version,
+                    "publisher_url": identity.publisher.publisher_url
+                    + "/commit/somesha"
```

Comment:
Extreme nitpick since it's a test, but let's do an f-string here: 

```python
f"{identity.publisher.publisher_url}/commit/somesha"
```

(Similarly for elsewhere below.)

Reply:
Gotcha. Given that it's just test code, this is fine then 🙂